### ISSUE TITLE
DwCACreator is regularly omitting taxonomic information

### ISSUE THREAD

**Issue by ansell** [AtlasOfLivingAustralia/biocache-store#205]
A review of the results of the GBIF archive regeneration job by @M-Nicholls and I showed that the taxonomic information was regularly being omitted by the DwCACreator algorithm. I am working on getting exact numbers today, but I am currently hampered #204 in some cases.

**Comment by ansell** [comment]
The statistics that came out of the data analysis for the GBIF archive files indicate there are serious issues around the taxonomic terms, including class, which is never populated:

fieldName | emptyCount | nonEmptyCount
---------- | ------------- | ----------------
http://rs.tdwg.org/dwc/terms/occurrenceID | 0 | 72308492
http://rs.tdwg.org/dwc/terms/catalogNumber | 20859451 | 51449041
http://rs.tdwg.org/dwc/terms/collectionCode | 46215151 | 26093341
http://rs.tdwg.org/dwc/terms/institutionCode | 42367185 | 29941307
http://rs.tdwg.org/dwc/terms/scientificName | 1371048 | 70937444
http://rs.tdwg.org/dwc/terms/recordedBy | 23293644 | 49014848
http://rs.tdwg.org/dwc/terms/taxonRank | 55266700 | 17041792
http://rs.tdwg.org/dwc/terms/kingdom | 50876004 | 21432488
http://rs.tdwg.org/dwc/terms/phylum | 63663753 | 8644739
http://rs.tdwg.org/dwc/terms/class | 72308492 | 0
http://rs.tdwg.org/dwc/terms/order | 55636881 | 16671611
http://rs.tdwg.org/dwc/terms/family | 46449647 | 25858845
http://rs.tdwg.org/dwc/terms/genus | 46605854 | 25702638
http://rs.tdwg.org/dwc/terms/specificEpithet | 48901521 | 23406971
http://rs.tdwg.org/dwc/terms/infraspecificEpithet | 70103920 | 2204572
http://rs.tdwg.org/dwc/terms/decimalLatitude | 5575563 | 66732929
http://rs.tdwg.org/dwc/terms/decimalLongitude | 5576884 | 66731608
http://rs.tdwg.org/dwc/terms/coordinatePrecision | 59012522 | 13295970
http://rs.tdwg.org/dwc/terms/coordinateUncertaintyInMeters | 25519462 | 46789030
http://rs.tdwg.org/dwc/terms/maximumElevationInMeters | 68071742 | 4236750
http://rs.tdwg.org/dwc/terms/minimumElevationInMeters | 66248185 | 6060307
http://rs.tdwg.org/dwc/terms/minimumDepthInMeters | 69621117 | 2687375
http://rs.tdwg.org/dwc/terms/maximumDepthInMeters | 69923339 | 2385153
http://rs.tdwg.org/dwc/terms/continent | 71349127 | 959365
http://rs.tdwg.org/dwc/terms/country | 35724810 | 36583682
http://rs.tdwg.org/dwc/terms/stateProvince | 33236875 | 39071617
http://rs.tdwg.org/dwc/terms/county | 61413642 | 10894850
http://rs.tdwg.org/dwc/terms/locality | 13956212 | 58352280
http://rs.tdwg.org/dwc/terms/year | 60752703 | 11555789
http://rs.tdwg.org/dwc/terms/month | 61081296 | 11227196
http://rs.tdwg.org/dwc/terms/day | 68112480 | 4196012
http://rs.tdwg.org/dwc/terms/basisOfRecord | 23004322 | 49304170
http://rs.tdwg.org/dwc/terms/identifiedBy | 64255183 | 8053309
http://rs.tdwg.org/dwc/terms/dateIdentified | 65792433 | 6516059
http://rs.tdwg.org/dwc/terms/occurrenceRemarks | 56582206 | 15726286
http://rs.tdwg.org/dwc/terms/locationRemarks | 68144196 | 4164296
http://rs.tdwg.org/dwc/terms/recordNumber | 63171497 | 9136995
http://rs.tdwg.org/dwc/terms/vernacularName | 18385067 | 53923425
http://rs.tdwg.org/dwc/terms/identificationQualifier | 71172405 | 1136087
http://rs.tdwg.org/dwc/terms/individualCount | 53043880 | 19264612
http://rs.tdwg.org/dwc/terms/eventID | 28891286 | 43417206
http://rs.tdwg.org/dwc/terms/geodeticDatum | 29184392 | 43124100
http://rs.tdwg.org/dwc/terms/eventTime | 63213513 | 9094979
http://rs.tdwg.org/dwc/terms/associatedSequences | 72308392 | 100
http://rs.tdwg.org/dwc/terms/eventDate | 6156294 | 66152198

I am going to look into working further through the existing issue at https://github.com/AtlasOfLivingAustralia/biocache-service/issues/128 to include the GBIF archives into the consolidation of all of the download implementations into a single code module, as the DwCACreator code with the bug in this case is difficult to grok and not well maintained.

**Comment by ansell** [comment]
I am regenerating the GBIF archives with the updated names and will rerun the analysis again after that is complete.

**Comment by ansell** [comment]
Looking at the initial results that are coming out of the regeneration job, it doesn't appear that there will be a change in the taxonomic field numbers.

The reason may be that the fields being used by the script are not the ".p" versions, so they may just be reproducing whatever was in the original data load:

https://github.com/AtlasOfLivingAustralia/biocache-store/blob/master/src/main/scala/au/org/ala/biocache/export/DwCACreator.scala#L83

@djtfmartin , was it intentional for the ALA to be sending GBIF unprocessed fields, particularly for the taxonomic information? 

If it was, we will need to spin up a further non-GBIF oriented set that contain the processed fields so that people can download data resources as Darwin Core Archives and have them look similar to the Biocache interface.

**Comment by djtfmartin** [comment]
@ansell yes it was. This was to avoid chinese whisper affect of classication matching.
